framework/src/org/checkerframework/framework/util/element/SuperTypeApplier.java

```diff
@@ -73,15 +73,16 @@ public static void annotateSupers(
      */
     @Override
     public int getElementIndex() {
-        return index;
+        return index == 0xffff ? -1 : index;
```

Comment:
Ah, this `index` is set by a private constructor of `SuperTypeApplier`. Current use case of this constructor doesn't directly read `TypeAnnotationPosition.type_index`. I will roll back this change.